spring-cloud-commons/src/main/java/org/springframework/cloud/client/discovery/AbstractDiscoveryLifecycle.java

```diff
@@ -35,7 +37,7 @@
  * Lifecycle methods that may be useful and common to various DiscoveryClient implementations.
```

Comment:
is it still various? or is registration (the `R` type) intended to work only with one sort of client now? I'd probably doc the type param regardless of whether it stays a marker or not.